### DISCUSSION TITLE
Add a dependency during projectless jack-in

### DISCUSSION THREAD

**Discussion by kxygk** [clojure-emacs/cider#3025] [Q&A]
When I launch a REPL without a project, I'd like to automatically add `org.clojure/tools.deps.alpha` so I can use `add-libs` and add dependencies dynamically

What I've done is add this to `~/.clojure/deps.edn` :

``` edn
 :deps { org.clojure/tools.deps.alpha {:git/url "https://github.com/clojure/tools.deps.alpha.git"
                                       :sha "241cd24c35ba770aea4773ea161d45276e5d3a73"}}
```

And this works!

But it's probably better to turn this off somehow if I'm launching CIDER in a project with it's own `deps.edn`. I'm just not clear on how to set a `:deps` only for non-project CIDER sessions. 

Maybe it all looks the same from the Clojure side and I need to modify something on the CIDER side - but I haven't been able to find the relevant bit in the documentation (or poking around with `C-h v`)

**Comment by lverns** [comment]
The easiest way to do this is to put that dependency under an alias. 

So `~/.clojure/deps.edn` might look something like

```edn
{:aliases {:projectless {:extra-deps { org.clojure/tools.deps.alpha {:git/url "https://github.com/clojure/tools.deps.alpha.git"
                                                                     :sha "241cd24c35ba770aea4773ea161d45276e5d3a73"}}}}}
```

Then you can include that alias when launching your repl: `-A:projectless`.

If you're connecting via cider-jack-in, press `C-u` before running cider-jack-in, so that you can add `-A:projectless` to the command.

This isn't the automatic detection you asked for, but it does allow you to manually choose at launch time. Also see the docs for the `cider-clojure-cli-global-options` if you want the alias to be used by default.
